src/lib/FlightTasks/tasks/Auto/FlightTaskAuto.cpp

```diff
@@ -300,6 +304,45 @@ void FlightTaskAuto::_updateAvoidanceWaypoints()
 	_desired_waypoint.waypoints[vehicle_trajectory_waypoint_s::POINT_2].point_valid = true;
 }
 
+void FlightTaskAuto::_checkAvoidanceProgress()
+{
+	position_controller_status_s pos_control_status = {};
+	pos_control_status.timestamp = hrt_absolute_time();
+
+	// vector from previous triplet to current target
+	Vector2f prev_to_target = Vector2f(&(_triplet_target - _triplet_prev_wp)(0));
```

Comment:
I think we should stop doing this with vectors and stick to value semantics (ie don't use &) even if it's slightly less compact.